### PR TITLE
fix: improve error message if custom build output is not found

### PR DESCRIPTION
**Pull request by petebacondarwin**
The message you get if Wrangler cannot find the output from the custom build is not more helpful.
It will even look around to see if there is a suitable file nearby and make suggestions about what should be put in the `main` configuration.

Closes [#946](https://github.com/cloudflare/wrangler2/issues/946)

### CODE REVIEW

**Review by threepointone — COMMENTED** [review]
We should have a similar error even when it isn't a custom build, so the error doesn't have to be specific to custom builds. We could add this error in getEntry (after the `fileExists(file)` check). What do you think?

File: packages/wrangler/src/entry.ts [hunk 1]
```diff
@@ -106,12 +106,31 @@ export async function runCustomBuild(
     });
 
     if (fileExists(expectedEntry) === false) {
-      throw new Error(
-        `Could not resolve "${path.relative(
-          process.cwd(),
-          expectedEntry
-        )}" after running custom build: ${build.command}`
-      );
+      const relativePath = path.relative(process.cwd(), expectedEntry) || ".";
```
Comment (JacobMGEvans) [inline packages/wrangler/src/entry.ts:109]: We should revisit the pathing PR again soon.

**Review by JacobMGEvans — APPROVED** [review]

**Comment by JacobMGEvans** [comment]
> We should have a similar error even when it isn't a custom build, so the error doesn't have to be specific to custom builds. We could add this error in getEntry (after the `fileExists(file)` check). What do you think?

I think it would be a good addition to non-custom builds to have a similar error. Especially since the functionality is very similar. Unless there is a lot that can be abstracted and shared between the two implementations it **could** be another PR. Tentatively Approved

**Comment by petebacondarwin** [comment]
I pulled the message generation out into a separate function and reused it for non-custom missing entry-points.
PTAL.

**Pull request merged**